sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffleCoalesceExec.scala

```diff
@@ -151,9 +151,6 @@ class GpuShuffleCoalesceIterator(
   }
 
   private def concatenateBatch(): ColumnarBatch = {
-    // about to start using the GPU in this task
-    GpuSemaphore.acquireIfNecessary(TaskContext.get(), semWaitTime)
```

Comment:
The new code no longer acquires the semaphore when there's no columns as it did before.  Is that a change we really want to make here?  I can think of cases where we could end up exploding the row-count-only batch into GPU data without holding the semaphore.

Reply:
Oh right that's a great catch. We should add an unconditional acquire for the empty column case.

Reply:
Handled here: https://github.com/NVIDIA/spark-rapids/pull/4396/commits/bc2a56cc55789ddeb9c8b516d3506f0057b3a9ad. I ran a few pointed queries that saw benefits with this new commit (like q94), I don't see it negate the benefit of the original patch, so I think it's still a go.